.github/workflows/release.yml

```diff
@@ -0,0 +1,61 @@
+name: Release
+on:
+  release:
+    types:
+      - prereleased
+      - released
+
+env: 
+ 
+  DOCKER_IMAGE: newrelic/nri-kube-events
+  RELEASE_VERSION: ${{ github.event.release.tag_name }}
+  S3_PATH: s3://nr-downloads-main/infrastructure_agent
+  PRERELEASE: ${{ github.event.release.prerelease }}
+
+jobs:
+  publish:
+    name: Publish
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Extract version from release tag
+        run: |
+          echo "RELEASE_VERSION=$(echo $RELEASE_VERSION | grep -oE '([0-9.]*[0-9])$')" >> $GITHUB_ENV
+
+      - name: Test for prerelase
+        if: ${{ github.event.release.prerelease }}
+        run: |
+          echo "RELEASE_VERSION=$RELEASE_VERSION-pre" >> $GITHUB_ENV
+          echo "S3_PATH=$S3_PATH/test" >> $GITHUB_ENV
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1 
+        with:
+          username: ${{ secrets.FSI_DOCKERHUB_TOKEN }}
+          password: ${{ secrets.FSI_DOCKERHUB_USERNAME }}
+
+      - name: Build and push
+        env:
+          DOCKER_IMAGE_NAME: ${{ env.DOCKER_IMAGE }}:${{ env.RELEASE_VERSION }}
+        run: |
+          make docker-build
+          docker push $DOCKER_IMAGE_NAME
+
+          # push 'latest' if NOT pre-release
+          if [[ -z "$PRERELEASE" || "false" == "$PRERELEASE" ]]; then
```

Comment:
I would do it as [this](https://github.com/newrelic/k8s-metadata-injection/pull/73/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R42) ,just to uniform the pipelines.